cogs/librarian/librarian.py

```diff
@@ -20,10 +20,17 @@ def __init__(self, bot):
         self.config = CogConfig("librarian")
         self.text = CogText("librarian")
 
+    async def fetch_data(self, url: str):
```

Comment:
This function should probably be called `fetch_json`, because of the `.json()` call on the return line. You should also use function annotations:
```py
async def fetch_json(self, url: str) -> dict:
```

We have a rule that non-command functions are at the bottom of the file, please move it there, below the `hash()` function.